### PR TITLE
Java-84 resolveAliases in TransactionService

### PR DESCRIPTION
**Pull request by fboucquez**
Hi guys,

This is my implementation of https://github.com/nemtech/nem2-sdk-java/issues/171. I've added the resolveAliases to the TransactionService

This PR includes another open PR https://github.com/nemtech/nem2-sdk-java/pull/170 as it's work on top of the TransactionService.

The implementation is different from the TS one as all the resolve aliases per transaction is in the TransactionServiceImpl instead of being spread on the Transaction hierarchy. I prefer to keep the transaction clean and closed for modification as much as possible.

What do you think?

### CODE REVIEW

File: sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,481 @@
...
+        return getResolvedValue(unresolvedAddress,
+            statementObservable.map(Statement::getAddressResolutionStatements),
+            expectedReceiptSource);
+    }
+
+    private <R, U> Observable<R> getResolvedValue(U unresolved,
+        Observable<List<? extends ResolutionStatement<U, R>>> resolutionsObservable,
+        ReceiptSource expectedReceiptSource) {
+
+        Predicate<ResolutionEntry<R>> sourceFilter = entry ->
```
Comment (dgarcia360) [inline sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java]: I'm not sure this ``entry.getReceiptSource().getPrimaryId() == expectedReceiptSource.getPrimaryId()`` is working as expected:

 -   Based on the example below, 2 transactions (index 0 & 1) are created on the same block, however, only 1 resolution entry get generated for both.

```
*  1 - Bob -> 1 random.token -> Alice
 * 2 - Carol -> 1 random.token > Denis

```
- Also, take into account that within the same block, a resolution can change:

```
*  1 - Bob -> 1 random.token (A) -> Alice
*  2 - Carol -> 1 random.token (A) > Denis
* 3 - Aggrgetate(MosaicAliasUnlink(random.token, A), MosaicAliasLink(random.token, B)
* 4 -  Bob -> 1 random.token (B) -> Alice
```

Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java]: The owner of the ResolutionEntry list is ResolutionStatement. ResolutionStatement contains the unresolved address/mosaic namespace. The filtering first selects the right ResolutionStatement for the address/mosaic namespace to be resolved and it walks thought ResolutionEntry list trying to select the right one based on the source's primary and secondary id. 

@dgarcia360 do you think that it's possible that ResolutionStatement has a source for a different unresolved value (alias)?

**Comment by fboucquez** [comment]
Hi @rg911 @dgarcia360 

I've just updated the resolution statement alias resolution according to the latest typescript.

Please have a look when you have the time.

File: sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java (hunk 1 above)
Comment (fboucquez) [reply sdk-core/src/main/java/io/nem/sdk/infrastructure/TransactionServiceImpl.java]: I've updated the primary, secondary id resolution according to TS.

**Pull request merged**